src/AzureAD.tsx

```diff
@@ -117,27 +126,8 @@ class AzureAD extends React.Component<IAzureADProps, IState> {
       return;
     }
 
-    this.dispatchLogout();
     this.authProvider.logout();
   };
-
-  private dispatchLogout = () => {
```

Comment:
The component should not be responsible for dispatching redux actions based on the changes in the authentication provider. Refactored this out so the provider becomes responsible. The allows us to build additional components that react to the same redux states.